trust/test-token.c

```diff
@@ -237,26 +267,30 @@ static void
 test_not_writable (void)
 {
 	p11_token *token;
+	char *path;
+	int fd;
 
-#ifdef OS_UNIX
-	if (getuid () != 0) {
-		token = p11_token_new (333, "/", "Label", P11_TOKEN_FLAG_NONE);
-		assert (!p11_token_is_writable (token));
-		p11_token_free (token);
-	}
-#endif
+	path = p11_path_build (test.unwritable, "test", NULL);
+	fd = open (path, O_WRONLY | O_CREAT, S_IRUSR | S_IWUSR);
+	free (path);
+	close (fd);
+
+	if (fd >= 0)
+		assert_skip ("cannot perform non-writable test", NULL);
```

Comment:
I suppose it does not matter that you leave the test file created here and that it will be removed in the teardown_temp() call?

Reply:
Yes, the test directory is recursively removed.